runtime/drivers/duckdb/model_executor_sqlstore_self.go

```diff
@@ -0,0 +1,101 @@
+package duckdb
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/rilldata/rill/runtime/drivers"
+)
+
+type sqlStoreToSelfInputProps struct {
+	SQL string `mapstructure:"sql"`
+	DSN string `mapstructure:"dsn"`
+}
+
+func (p *sqlStoreToSelfInputProps) Validate() error {
+	if p.SQL == "" {
+		return fmt.Errorf("missing property 'sql'")
+	}
+	if p.DSN == "" {
+		return fmt.Errorf("missing property `dsn`")
+	}
```

Comment:
The DSN or other connection properties (like `host`, etc.) may have been configured in a `postgres.yaml` file instead of inline. You should be able to access that info through `ModelExecutorOptions.InputHandle`.

Reply:
We will need to create a model and connector both from a source definition.

Reply:
But it will be easy to rewrite sources to models if allow inline properties. I have split opinions on this.
What do you suggest ?

Reply:
I'm also torn on this, I will generally favor the overall simplest solution, keeping in mind that we must keep backwards compatibility for existing sources.

One idea I can think of is to use nested values in `connector:` in models for inline-defined connectors. And then for backwards compatibility with sources, moving the few known properties like `dsn:` and similar to it. Specifically, the YAML would be:
```type
type: model
connector:
  driver: postgres
  dsn: psql://....
...
```

This should then emit a connector named `<model_name>__<connector>` and set it as the model's `InputConnector`. 

Would this work? Again, it's just a high-level idea and I'm overall biased towards not having any solutions here that get too complex/weird.